test-integ/peering_commontopo/setup.go

```diff
@@ -0,0 +1,58 @@
+package peering
+
+import (
+	"flag"
+	"testing"
+)
+
+// Test that use commonTopo should implement commonTopoSuite.
+//
+// Tests that use commonTopo are either cooperative or non-cooperative. Non-cooperative
+// uses of commonTopo include is anything that may interfere with other tests, namely
+// mutations, such as:
+// - any calls to commonTopo.Relaunch; this is generally disruptive to other tests
+// - stopping or disabling nodes
+// - ...
+//
+// Cooperative tests should just call testFuncMayReuseCommonTopo() to ensure they
+// are run in the correct `sharetopo` mode. They should also ensure they are included
+// in the commonTopoSuites slice in TestSuitesOnSharedTopo.
+type commonTopoSuite interface {
+	testName() string
+	setup(*testing.T, *commonTopo)
+	test(*testing.T, *commonTopo)
+}
+
+var commonTopologyFlag = flag.Bool("commontopo", false, "run tests with common topology")
+
+func commonTopologyTest(t *testing.T) {
+	t.Helper()
+	if *commonTopologyFlag {
+		t.Skip(`Skipping: duplicate test run. See "TestSuitesOnSharedTopo"`)
+	}
+}
+
+func setupAndRunTestSuite(t *testing.T, suites []commonTopoSuite, shareTopology, runParallel bool) {
+	if shareTopology {
+		commonTopologyTest(t)
+	}
+
+	if runParallel {
```

Comment:
I think it is incorrect to control both the inter-suite parallelism and the `s.test` parallelism with one flag. Inter-suite is bound by CPU/RAM and other system resources and IME should be disabled. `s.test` parallelism is generally fine for non-destructive tests.

Reply:
I pointed this out in my comment here https://github.com/hashicorp/consul/pull/18132/files#r1264096291 BTW

Reply:
I must have missed the comment, I'll update it in the next PR with the README update. Cheers